tensorflow/training/docker/2.5/py3/cu112/Dockerfile.gpu

```diff
@@ -0,0 +1,327 @@
+FROM nvidia/cuda:11.2.1-base-ubuntu18.04
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# prevent stopping by user interaction
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
+
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+
+# Set environment variables for MKL
+# For more about MKL with TensorFlow see:
+# https://www.tensorflow.org/performance/performance_guide#tensorflow_with_intel%C2%AE_mkl_dnn
+ENV KMP_AFFINITY=granularity=fine,compact,1,0
+ENV KMP_BLOCKTIME=1
+ENV KMP_SETTINGS=0
+ENV RDMAV_FORK_SAFE=1
+
+ARG PYTHON=python3.7
+ARG PYTHON_PIP=python3-pip
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.7.10
+ARG OPENSSL_VERSION=1.1.1g
+ARG OPEN_MPI_PATH=/opt/amazon/openmpi
+ARG EFA_PATH=/opt/amazon/efa
+ARG EFA_VERSION=1.11.2
+ARG OMPI_VERSION=4.1.1
+ARG BRANCH_OFI=1.1.2
+
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r2.5_aws/20210515_165343/gpu/cu112/py37/tensorflow_gpu-2.5.0-cp37-cp37m-manylinux2010_x86_64.whl
+
+ARG ESTIMATOR_URL=
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=1.0.9
+
+ARG CUDA=11.2
+ARG CUDA_DASH=11-2
+ARG CUDNN=8.1.0.77-1
+ARG NCCL_VERSION=2.8.4
+ARG CUDNN_MAJOR_VERSION=8
+ARG LIB_DIR_PREFIX=x86_64
+
+
+RUN apt-get update && apt-get install -y --no-install-recommends --allow-unauthenticated --allow-downgrades  --allow-change-held-packages \
+   ca-certificates \
+   cuda-command-line-tools-${CUDA_DASH} \
+   cuda-cudart-dev-${CUDA_DASH} \
+   libcufft-dev-${CUDA_DASH} \
+   libcurand-dev-${CUDA_DASH} \
+   libcusolver-dev-${CUDA_DASH} \
+   libcusparse-dev-${CUDA_DASH} \
+   curl \
+   emacs \
+   libcudnn8=${CUDNN}+cuda${CUDA} \
+   # TensorFlow doesn't require libnccl anymore but Open MPI still depends on it
+   libgomp1 \
+   libfreetype6-dev \
+   libhdf5-serial-dev \
+   liblzma-dev \
+   libpng-dev \
+   libtemplate-perl \
+   libzmq3-dev \
+   git \
+   unzip \
```

Comment:
EFA dependencies also include `hwloc` but again its missing from this file -> causing smddp runs to fail on p3dn

Refer TF2.4 gpu dockerfile : 
https://github.com/aws/deep-learning-containers/blob/f9ee2faee183fa87049efb892648714e0d27fe68/tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu#L68

PR https://github.com/aws/deep-learning-containers/pull/1044 introduced it

Reply:
done